ui/components/multichain/pages/send/send.js

```diff
@@ -230,20 +233,30 @@ export const SendPage = () => {
     event.preventDefault();
 
     setIsSubmitting(true);
-    await dispatch(signTransaction(history));
-    // prevents state update on unmounted component error
-    if (isSubmitting) {
+    setError(undefined);
+
+    try {
+      await dispatch(signTransaction(history));
+
+      // prevents state update on unmounted component error
+      if (isSubmitting) {
+        setIsSubmitting(false);
+      }
+
+      trackEvent({
+        category: MetaMetricsEventCategory.Transactions,
+        event: 'Complete',
+        properties: {
+          ...sendAnalytics,
+          action: isSwapAndSend ? 'Submit Immediately' : 'Edit Screen',
+          legacy_event: true,
+        },
+      });
+    } catch {
       setIsSubmitting(false);
+      setError(TRANSACTION_ERRORED_EVENT);
+      // throw error;
     }
```

Comment:
Minor: might be clearer to call `setIsSubmitting(false)` in a finally block

Reply:
True